packages/react-text/src/components/Text/useTextStyles.ts

```diff
@@ -17,7 +104,33 @@ const useStyles = makeStyles({
  */
 export const useTextStyles = (state: TextState): TextState => {
   const styles = useStyles();
-  state.className = mergeClasses(styles.root, state.className);
+  state.className = mergeClasses(
+    styles.root,
+    state.wrap === false && styles.nowrap,
+    state.truncate === true && styles.truncate,
+    state.block === true && styles.block,
+    state.italic === true && styles.italic,
+    state.underline === true && styles.underline,
+    state.strikethrough === true && styles.strikethrough,
+    state.underline === true && state.strikethrough === true && styles.strikethroughUnderline,
```

Comment:
```suggestion
    state.truncate && styles.truncate,
    state.block && styles.block,
    state.italic true && styles.italic,
    state.underline && styles.underline,
    state.strikethrough && styles.strikethrough,
    state.underline && state.strikethrough && styles.strikethroughUnderline,
```

_nit:_ is there need in strict comparisons?

Reply:
The reasoning behind the strict comparisons was to make readability better for figuring out the condition vs the value used.
Is there any benefit to the _truthy_ alternative?

Reply:
in general using short circuiting in JS is an anti-pattern and should be avoided:
- it can lead to unexpected results (`true && -10` will produce `-10` etc )
- non explicit code / extensive cognitive load for human to process the intent

because `mergeClasses` handles most of this behaviour under the hood (except `number` value 🚨 ), I'd say it's "safe" to use short-circuiting (although it produces non explicit code as mentioned).

- while your strict comparisons is more readable it doesn't add additional value unfortunately. To be explicit the right way, you'd need to use ternary operator.
- also `state.truncate && styles.truncate` will produce smaller payload (I think) than `state.truncate === true && styles.truncate `

Reply:
Thanks for the explanation Martin. Given that `mergeClasses` already handles the behavior, we're going with @layershifter's suggestion

Reply:
Should we also go with this:
```js
!state.wrap && styles.nowrap,
```
Over this?
```js
state.wrap === false && styles.nowrap,
```

Reply:
hmm so negatives are even more hard to read... I'd stay with explicit `state.wrap === false && styles.nowrap,` for this one